File: NeetCodeIO/1448CountGoodNodesInBinaryTree/CountGoodNodesInBinaryTree.cpp
class Solution {
public:
    int ans = 0;
    void countGoodNodes(TreeNode* node, TreeNode* parent) {
        if (!node) return;
        if (!parent || node->val >= parent->val) {
            ans++;
            parent = node;
        }
        countGoodNodes(node->left, parent);
        countGoodNodes(node->right, parent);
    }
    int goodNodes(TreeNode* root) {
        // DFS -> Time:O(n), Space:O(n)
        countGoodNodes(root, NULL);
        return ans;
    }
};

File: NeetCodeIO/1448CountGoodNodesInBinaryTree/CountGoodNodesInBinaryTree.py
class Solution:
    ans = 0
    def countGoodNodes(self, node, parent):
        if not node:
            return 
        if not parent or node.val >= parent.val:
            self.ans += 1
            parent = node
        self.countGoodNodes(node.left, parent)
        self.countGoodNodes(node.right, parent)

    def goodNodes(self, root: TreeNode) -> int:
        # DFS -> Time:O(n), Space:O(n)
        self.countGoodNodes(root, None)
        return self.ans
